catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/FromIcebergType.java

```diff
@@ -26,7 +26,16 @@ public Type schema(Schema schema, Type structType) {
 
   @Override
   public Type struct(Types.StructType struct, List<Type> fieldResults) {
-    throw new UnsupportedOperationException("Data conversion of struct type is not supported");
+    return com.datastrato.gravitino.rel.types.Types.StructType.of(
+        struct.fields().stream()
```

Comment:
It's cool to use stream API, but I prefer to use for iterate to avoid `struct.fields().indexOf(nestedField)` in case of big structs.
https://github.com/apache/iceberg/blob/main/spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/TypeToSparkType.java#L62-L71

Reply:
I'm ok for the current implement, it deponds on you :)

Reply:
Could you provide a detailed explanation of the benefits of iterate implementation? 
What negative impact will be caused if the structs is too big?

Reply:
`struct.fields().indexOf(nestedField)` is O(n)

Reply:
Thanks, I'll try to implement by iterate